### DISCUSSION TITLE
Is limiting relationships possible?

### DISCUSSION THREAD

**Discussion by adammsteele** [dabapps/django-readers#67] [Q&A] [answered]
Hi,

Just trying to get my head around what is and isn't supported, the use case I'm after would be something like an `Author` with many `Book`s, and writing a spec to show the author and the top 5 books by some ordering criteria.

I tried a couple of approaches to no avail:

- I wrote a `slice` pair which used `QuerySet.__getitem__` to mimic slicing, but I think that won't work in general if there are other filters to be applied to the overall queryset
- An annotation using `row_number` also did not work because one cannot filter on a window function

Let me know if you would prefer questions to be asked on a different platform.

Thanks for the library.

**Comment by j4mie** [comment, marked as answer]
Hi! Thanks for trying out django-readers 😎 

I think the question isn't really readers-specific: readers just uses `prefetch_related` to load relationships, so what you need to be able to do is prefetch a limited number of objects. [This article](https://stackoverflow.com/questions/24690755/django-prefetch-related-with-limit) suggests a few ways to do it, involving `OuterRef` and `Subquery`. If you can get this to work with "just" your queryset (ie without readers) then it should be fairly straightforward to convert this into a reader pair.

I'll leave this ticket open because I'm very interested to hear how you get on!

**Comment by j4mie** [comment]
Looks like this is going to be possible in Django 4.2: https://docs.djangoproject.com/en/dev/releases/4.2/#models

> [prefetch_related()](https://docs.djangoproject.com/en/dev/ref/models/querysets/#django.db.models.query.QuerySet.prefetch_related) now supports [Prefetch](https://docs.djangoproject.com/en/dev/ref/models/querysets/#django.db.models.Prefetch) objects with sliced querysets.

cc @simkimsia (re #73)
